### PR TITLE
(DOCS-4111) Re-organize NDM left nav and rename setup page

### PR DESCRIPTION
**Pull request by jtappa**
### What does this PR do?
This re-organizes the NDM Devices section to put SNMP Traps immediately under the SNMP Metrics (re-named from "Setup") page. This better aligns what these pages are actually about, as setting up SNMP Metrics isn't required to use other forms of NDM (but it is encouraged)

### Motivation
Request from PM, and prepping for Dash.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/network_monitoring/devices/guide/tags-with-regex.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Datadog Network Device Monitoring (NDM) supports regular expressions to create m
 
 ### Installation
 
-Follow the [setup instructions][1] to install Datadog NDM.
+Follow the [setup instructions][1] to install Datadog NDM SNMP metrics.
```
Comment (naaltman) [inline content/en/network_monitoring/devices/guide/tags-with-regex.md:21]: I think we should leave this as "install Datadog Network Device Monitoring"

File: content/en/network_monitoring/devices/snmp_metrics.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 ---
-title: Network Device Monitoring Setup
+title: SNMP Metrics
```
Comment (naaltman) [inline content/en/network_monitoring/devices/snmp_metrics.md:2]: Change title to "Collect SNMP Metrics from your Devices" 

This is still the general setup for the product, e.g you must do this to get any of the UI to work, but I want to be clear that this setup does not include sending Traps or Netflow hence the separation, so I think it's best to add more to the tile

Comment (jtappa) [inline content/en/network_monitoring/devices/snmp_metrics.md:2]:
```suggestion
title: Collect SNMP Metrics from your Network Devices
```

File: content/en/network_monitoring/devices/snmp_metrics.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
 ---
-title: Network Device Monitoring Setup
+title: Collect SNMP Metrics from your Network Devices
```
Comment (jtappa) [inline content/en/network_monitoring/devices/snmp_metrics.md:2]:
```suggestion
title: Collect SNMP Metrics From Your Network Devices
```

**Review by alai97 — APPROVED** [review]

**Pull request merged**